src/ResourceManagement/ResourceManager/GenericResource/GenericResourceImpl.cs

```diff
@@ -143,9 +144,10 @@ public GenericResource.Definition.IWithPlan WithProviderNamespace(string resourc
             return this;
         }
 
-        public IWithCreate WithApiVersion(string apiVersion)
+        public IWithCreate WithApiVersion(string apiVersion = default(string))
         {
-            this.apiVersion = apiVersion;
+            this.apiVersion = apiVersion ?? ResourceUtils.ApiVersionFromResourceProviderAndType(resourceProviderNamespace, resourceType);
```

Comment:
Just a thought - Now that we have the list of supported versions, could we validate the specified api version to see if it's supported or not?

Reply:
The list of supported versions come from the clients of current SDK. It could be used when users have no info about the api versions and such type of resource gets supported in our SDK. But there are wide range of resources/api versions in Azure. It's hard to validate specified api version now.